weave/trace_server_bindings/remote_http_trace_server.py

```diff
@@ -148,22 +171,25 @@ def _flush_calls(
 
         try:
             self._send_batch_to_server(encoded_data)
-        except Exception:
-            # Add items back to the queue for later processing
-            logger.warning(
-                f"Batch failed after max retries, requeueing batch with {len(batch)=} for later processing",
-            )
+        except Exception as e:
+            if not _is_retryable_exception(e):
+                _log_dropped_call_batch(batch, e)
```

Comment:
Suppose a batch contains 10 calls, and only 1 call raises an non-retry-able error.  Won't the code as written drop all 10 calls?  Shouldn't we just drop the 1 call that is not retry-able?

Reply:
Yes but the server doesn't tell us which call made the request bad. We could bisect the batch and try each half but that will mean n log n server calls to potentially upload zero calls in the worst case, if all the calls in the batch are bad

Reply:
Also this PR does not modify that behavior- currently if you have a batch of 10 calls the behavior is we retry the batch forever and drop all the calls because every attempt 400s

Reply:
hmm okay I think this is still better than what we have today, but please make a note to revisit this